### DISCUSSION TITLE
C++ virtualcam

### DISCUSSION THREAD

**Discussion by shersoni610** [letmaik/pyvirtualcam#55] [Q&A]
Hello

It seems that the code uses pybind for the python. I was wondering if it is possible to use this software directly within c++ codes? I do not know how to combine c++ codes with *.mm codes.

Thanks

**Comment by justaquickhack** [comment]
Hello,

I have been using pyvirtualcam to prototype some OpenCV filters for a webcam overlay on Windows and it is working great.  Thank you @letmaik for all of your hard work on this!

I would also like to migrate that to C++ for several reasons.  Unfortunately, I am unable to find anything for doing this in C++ that is as easy to use as pyvirtualcam.  There are some C++ examples out there for Windows, but they seem to require a lot of DirectShow/Graph knowledge.  How hard would it be to implement this directly in C++ without the pybind?  In my searches, it seems like there are quite a few people looking for this.  Any examples or links to existing code would be greatly appreciated.

Thanks

**Reply by letmaik** [reply]
Did you look at the C++ code of the backends of this package? There is a clear separation between the Python binding (using pybind11) and the actual backend, which should be reusable in other C++ code.

**Reply by shersoni610** [reply]
Hello,

May I kindly request you for some help on how to use your software in C++ applications? Is there a CMakeLists.txt file that can
be used to build this software as a library?

Regards

**Comment by shersoni610** [comment]
Hello,

In order to use the software in C++, I created the following Cmakelist in native_mac_os folder. However, I get the following error:

error: ARC forbids explicit message send of 'dealloc'
        [_mach_server dealloc];

can someone help?
csv


cmake_minimum_required(VERSION 3.19)
set(CMAKE_CXX_STANDARD_REQUIRED ON)

set(CMAKE_CXX_STANDARD 17)
project(vircam)

set(CMAKE_C_FLAGS "-x objective-c")
set(CMAKE_CXX_FLAGS "-w -x objective-c++")
set(CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_RELEASE} -Ofast")

set(CMAKE_EXE_LINKER_FLAGS "-framework Cocoa -framework AppKit -framework CoreData -framework Foundation -framework AVFoundation -framework CoreVideo -framework CoreMedia -w")

IF (APPLE)
        set(OBJC_SRC main.mm ./server/OBSDALMachServer.mm)
        set_source_files_properties(${OBJC_SRC} PROPERTIES COMPILE_FLAGS -fobjc-arc)
ENDIF ()

add_executable(vircam ${OBJC_SRC})
include_directories(vircam ../../external/libyuv/include)
